pkg/cassandra/config.go

```diff
@@ -367,12 +368,24 @@ func filterConfigForVersion(cassandraVersion string, cassandraYamlIR *CassYamlIR
 
 // ApplySystemReplication adds system properties to configure replication of system
 // keyspaces.
-func ApplySystemReplication(dcConfig *DatacenterConfig, replication SystemReplication) {
-	dcNames := SystemReplicationDcNames + "=" + strings.Join(replication.Datacenters, ",")
-	replicationFactor := SystemReplicationFactor + "=" + strconv.Itoa(replication.ReplicationFactor)
+func ApplySystemReplication(dcConfig *DatacenterConfig, replication map[string]int) {
+	replicationFactors := make([]string, 0, len(replication))
+	dcs := make([]string, 0, len(replication))
+
+	// Sort to make verification in tests easier.
+	for k := range replication {
+		dcs = append(dcs, k)
+	}
+	sort.Strings(dcs)
+
+	for _, dc := range dcs {
+		replicationFactors = append(replicationFactors, fmt.Sprintf("%s:%d", dc, replication[dc]))
+	}
+	replicationFactor := SystemReplicationFactor + "=" + strings.Join(replicationFactors, ",")
```

Comment:
Same here, `replicationFactor` is misleading as to what this variable contains. `systemReplicationStrategy` or `systemReplication` would be more accurate.